### PR TITLE
[Misc] Add metrics for request queue time, forward time, and execute time

### PR DESCRIPTION
**Pull request by Abatom**
1. Add metrics for request queue time, forward time, and execute time, which can be returned through the `/metrics` API.
2. Remove the restriction on `collect_model_forward_time` and `collect_model_execute_time` due to the `--otlp-traces-endpoint` flag, so that metrics can also collect information about `model_forward_time` and `model_execute_time`.

### CODE REVIEW

**Comment by abatom** [comment]
@WoosukKwon, Could you help me review the code?

**Comment by abatom** [comment]
@njhill , Could you help me review the code? Thanks!

**Comment by Abatom** [comment]
@youkaichao , Could you help me review the code? Thanks!

File: vllm/engine/metrics.py [hunk 1]
```diff
@@ -133,7 +133,37 @@ def __init__(self, labelnames: List[str], max_model_len: int):
...
+            ])
+        self.histogram_model_execute_time_request = self._histogram_cls(
+            name="vllm:model_execute_time_seconds",
+            documentation=
+            "Histogram of time spent in the model execute function in seconds.",
+            labelnames=labelnames,
+            buckets=[
+                0.3, 0.5, 0.8, 1.0, 1.5, 2.0, 2.5, 5.0, 10.0, 15.0, 20.0, 30.0,
+                40.0, 50.0, 60.0
+            ])
```
Comment (simon-mo) [inline vllm/engine/metrics.py:166]: The model forward and execution typically runs in the order of ms not s. This histogram bucket is not every informative. I suggest (1) keep it unit of ms (2) bucket from 2ms to 200ms

**Comment by mgoin** [comment]
cc @robertgshaw2-neuralmagic

File: vllm/engine/metrics.py (hunk 1 above)
Comment (Abatom) [reply vllm/engine/metrics.py]: > The model forward and execution typically runs in the order of ms not s. This histogram bucket is not every informative. I suggest (1) keep it unit of ms (2) bucket from 2ms to 200ms

Thank you for your reply. In my actual testing, I found that the latency of forward and execute sometimes reaches 800ms, so I increased the value range from 1ms to 3000ms to accommodate different application scenarios.

**Review by simon-mo — APPROVED** [review]

**Pull request merged**